src/CloudNative.CloudEvents/CloudEventFormatters.cs

```diff
@@ -0,0 +1,205 @@
+// Copyright 2025 Cloud Native Foundation.
+// Licensed under the Apache 2.0 license.
+// See LICENSE file in the project root for full license information.
+
+using CloudNative.CloudEvents.Core;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Mime;
+using System.Threading.Tasks;
+
+namespace CloudNative.CloudEvents;
+
+/// <summary>
+/// Extension methods for <see cref="CloudEventFormatter"/>.
+/// </summary>
+public static class CloudEventFormatters
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="formatter"></param>
+    /// <param name="extensionAttributes"></param>
+    /// <returns></returns>
+    public static CloudEventFormatter WithDefaultDecodingExtensions(this CloudEventFormatter formatter, IEnumerable<CloudEventAttribute> extensionAttributes) =>
+        new DefaultExtensionsFormatter(formatter, extensionAttributes);
+
+    /// <summary>
+    /// 
+    /// </summary>
+    /// <param name="formatter"></param>
+    /// <param name="validationAction"></param>
+    /// <returns></returns>
+    public static CloudEventFormatter WithValidation(this CloudEventFormatter formatter, Action<CloudEvent> validationAction) =>
```

Comment:
I don't suppose it's strictly necessary but it looks like there's no builder pattern here. Not sure if we'd want an API that allows users to write code like:

```csharp
var formatter = new JsonEventFormatter()
  .WithRequiredTenantExtension()
  .WithEventFormatValidation()
```

Reply:
That's actually *exactly* the kind of code I'd expect them to write, to be honest. And that ends up with multi-layer delegation, but I think that's okay, given that they'll be implemented by separate classes.

Happy to understand any concerns you have though. We would want to clearly document the order in which this sort of "pipeline" is applied, of course...

Reply:
Disregard me -- I'm blind. We're chaining on the actual `CloudEventFormatter` here. No builder required.